sampleExtension/background.js

```diff
@@ -1,60 +1,117 @@
 
 //Set that holds the urls to be intercepted
 var blockedSites = [];
+var interceptCounter = 0;
+var interceptDateList = [];
+//Boolean that determines whether we should be redirecting or not
+var interceptionEnabled = true;
 
-initExtension = function () {
-    //First receive the blacklist from the sync storage, and then create a onBeforeRequest listener using this list.
-    updateBlockedSites(replaceListener);
-    addBrowserActionListener();
+/* --------------- ------ update list of BlockedSites ------ ---------------*/
+
+updateStorage = function() {
+    setStorageBlacklist(blockedSites);
+    return {redirectUrl: "https://zeeguu.herokuapp.com/get-ex"};
 };
 
 // This function receives the blacklist from the sync storage.
-updateBlockedSites = function(callback){
-    chrome.storage.sync.get("tds_blacklist", function (items) {
-        if (!chrome.runtime.error) {
-            blockedSites = items.tds_blacklist;
-            return callback();
-        }
+retrieveBlockedSites = function(callback){
+    getStorageBlacklist(function(blacklist) {
+        blockedSites = blacklist;
+        return callback();
+    });
+};
+
+//Loads the intercept time+date list from storage
+retrieveInterceptDateList = function() {
+    getInterceptDateList(function(dateList) {
+        interceptDateList = dateList;
+    });
+};
+
+//Loads the intercept counter from storage
+retrieveInterceptCounter = function(callback) {
+    getInterceptCounter(function(counter) {
+        interceptCounter = counter;
+        return callback();
     });
 };
 
 // This function adds one url to the blacklist
 addToBlockedSites = function(urlToAdd) {
-    var formatUrl = urlToAdd.split(['//'])[1];
-    formatUrl = '*://' + formatUrl + '*';
-    blockedSites.push(formatUrl);
-    chrome.storage.sync.set({"tds_blacklist" : blockedSites }, function() {
-        if(chrome.runtime.error) {
-            console.log("Runtime error.");
-        }
-    });
+    var blockedSiteItem = new BlockedSite(formatUrl(urlToAdd));
+    blockedSites.push(blockedSiteItem);
+    updateStorage();
+    replaceListener();
+};
+
+// This function adds the current time+date to the saved time+date list
+addToInterceptDateList = function() {
+    var newDate = new Date().toDateString();
+    if (interceptDateList == null) {
+        interceptDateList = [newDate];
+    } else {
+        interceptDateList.push(newDate);
+    }
+    setInterceptDateList(interceptDateList);
+};
+
+//TODO fix in interation 3 for url-handling module
+formatUrl = function(url) {
```

Comment:
The TODO here suggests that this will be gone in the future, which is the case. We are working on a url handling module as i write this